sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,241 @@ use crate::{
             }
         }
 
+        residents.shuffle(rng);
+        workers.shuffle(rng);
+
         let mut s = Scenario::empty(map, "random people going to/from work");
         // Include all buses/trains
         s.only_seed_buses = None;
 
-        for mut person in timer
-            .parallelize(
-                "create people",
-                Parallelism::Fastest,
-                residences,
-                |(home, mut forked_rng)| robot_person(home, &workplaces, map, &mut forked_rng),
-            )
-            .into_iter()
-            .flatten()
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
-        }
+        let residents_cap = residents.len();
+        let workers_cap = workers.len();
+
+        // this saturation figure is an arbitrary guess - we assume that the number of trips will
+        // scale as some factor of the people living and/or working on the map. A number of more
+        // than 1.0 will primarily affect the number of "pass through" trips - people who neither
+        // work nor live in the neighborhood.
+        let trip_saturation = 0.8;
+        let num_trips = (trip_saturation * (residents_cap + workers_cap) as f64) as usize;
 
-        // Create trips between map borders. For now, scale the number by the number of residences.
-        let incoming_connections = map.all_incoming_borders();
-        let outgoing_connections = map.all_outgoing_borders();
-        for mut person in timer
-            .parallelize(
-                "create border trips",
-                Parallelism::Fastest,
-                std::iter::repeat_with(|| abstutil::fork_rng(rng))
-                    .take(num_residences)
-                    .collect(),
-                |mut forked_rng| {
-                    border_person(
-                        &incoming_connections,
-                        &outgoing_connections,
-                        map,
-                        &mut forked_rng,
-                    )
-                },
+        // bound probabilities to ensure we're getting some diveristy of agents
+        let lower_bound_prob = 0.05;
+        let upper_bound_prob = 0.90;
+        let prob_local_resident = if workers_cap == 0 {
+            lower_bound_prob
+        } else {
+            f64::min(
+                upper_bound_prob,
+                f64::max(lower_bound_prob, residents_cap as f64 / num_trips as f64),
             )
-            .into_iter()
-            .flatten()
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
+        };
+        let prob_local_worker = f64::min(
+            upper_bound_prob,
+            f64::max(lower_bound_prob, workers_cap as f64 / num_trips as f64),
+        );
+
+        info!(
+            "BUILDINGS - workplaces: {}, residences: {}, mixed: {}",
+            num_bldg_commercial, num_bldg_residential, num_bldg_mixed_residential_commercial
```

Comment:
If you read these values frequently, `prettyprint_usize` makes it easier to understand larger values IMO